extra/kotlin/src/main/kotlin/org/spongepowered/configurate/kotlin/Processors.kt

```diff
@@ -0,0 +1,28 @@
+package org.spongepowered.configurate.kotlin
+
+import org.spongepowered.configurate.CommentedConfigurationNodeIntermediary
+import org.spongepowered.configurate.ConfigurationNode
+import org.spongepowered.configurate.objectmapping.meta.Comment
+import org.spongepowered.configurate.objectmapping.meta.Processor
+import java.lang.reflect.Type
+
+/**
+ * Apply comments from [Comment] annotation on save. This will first call the [String.trimIndent] method on the strings,
+ * allowing multi-line Kotlin strings in [Comment] annotations without worrying about indentation.
+ *
+ * @return a new processor factory
+ */
+fun kotlinComments(): Processor.Factory<Comment, Any?> {
```

Comment:
I'm not sure about this name since you don't have the file name as part of the import -- maybe something like `kotlinCommentsProcessor()`?

Reply:
Yeah that's fair enough, I wasn't sure about the name either. It's too bad Kotlin doesn't allow static extensions off of Java classes :(

Reply:
I've updated the PR, not sure how I feel about the name either way to be honest. Feels a bit clunky (so did just `kotlinComments()`), but it's definitely more clear. Will try to think of something better.

Reply:
I'm fine with clunky but clear for this sort of API -- it's only referenced when setting up an object mapper, not something that would spread widely across a codebase like node operations, so the impact of the name itself is more limited.